src/operator/mshadow_op.h

```diff
@@ -99,26 +117,35 @@ struct reciprocal_grad {
 struct sigmoid {
   template<typename DType>
   MSHADOW_XINLINE static DType Map(DType a) {
-    return DType(DType(1.0f) / (DType(1.0f) + expf(-a)));
+    float af(static_cast<float>(a));
+    return DType(1.0f / (1.0f + math::exp(-af)));
   }
 };
+
+template<>
+MSHADOW_XINLINE double sigmoid::Map<double>(double a) {
+  return 1.0 / (1.0 + math::exp(-a));
```

Comment:
I'll merge this for 0.12 release. But I think there is a cleaner way to do this that doesn't require specialization